src/platform_impl/macos/util/mod.rs

```diff
@@ -146,3 +147,17 @@ pub unsafe fn toggle_style_mask(window: id, view: id, mask: NSWindowStyleMask, o
     // If we don't do this, key handling will break. Therefore, never call `setStyleMask` directly!
     window.makeFirstResponder_(view);
 }
+
+pub unsafe fn id_to_string(string: id) -> String {
+    let has_attr: BOOL = msg_send![string, isKindOfClass: class!(NSAttributedString)];
+    let characters = if has_attr != NO {
+        // This is a *mut NSAttributedString
+        msg_send![string, string]
+    } else {
+        // This is already a *mut NSString
+        string
+    };
+    let utf8_sequence =
+        std::slice::from_raw_parts(characters.UTF8String() as *const c_uchar, characters.len());
+    std::str::from_utf8_unchecked(utf8_sequence).to_string()
```

Comment:
Good point. We don't know how `UTF8String` handles strings that are already ill-formed.

I'd suggest using `String::from_utf8_lossy` because panicking in such cases would be uncalled for. In fact we should also rename the function to `id_to_string_lossy` to be explicit about this, and to allow for creating another version of this function which returns a Result in case the conversion failed.

Reply:
This is now addressed